src/test/scala/io/iohk/ethereum/jsonrpc/server/http/JsonRpcHttpServerSpec.scala

```diff
@@ -32,20 +33,44 @@ class JsonRpcHttpServerSpec extends AnyFlatSpec with Matchers with ScalatestRout
   it should "respond to healthcheck" in new TestSetup {
     (mockJsonRpcHealthChecker.healthCheck _)
       .expects()
-      .returning(Task.now(HealthcheckResponse(List(HealthcheckResult("listening", "OK", None)))))
+      .returning(Task.now(HealthcheckResponse(List(HealthcheckResult.ok("peerCount", Some("2"))))))
 
     val getRequest = HttpRequest(HttpMethods.GET, uri = "/healthcheck")
 
     getRequest ~> Route.seal(mockJsonRpcHttpServer.route) ~> check {
       status shouldEqual StatusCodes.OK
-      responseAs[String] shouldEqual """{
+      parse(responseAs[String]) shouldEqual parse("""{
                                        |  "checks":[
-                                       |    {
-                                       |      "description":"listening",
-                                       |      "status":"OK"
-                                       |    }
+                                       |    { "name": "peerCount", "status": "OK", "info": "2" }
                                        |  ]
-                                       |}""".stripMargin
+                                       |}""".stripMargin)
+    }
+  }
+
+  it should "respond to healthcheck with an error if one healtcheck fails" in new TestSetup {
```

Comment:
Can you add tests for the other checks as well? Not just peerCount

Reply:
And a typo in `if one healtcheck fails`

Reply:
Actually, it does not really check `peerCount` itself because the checks come from a mock. The only logic that is tested here is the resulting json and the http code which should be 500 as soon as one check fails. I don't think that adding other tests with the `mockJsonRpcHealthChecker` really makes sense.

Reply:
👍